src/components/partnership/partnership.service.ts

```diff
@@ -365,32 +358,26 @@ export class PartnershipService {
     const object = await this.readOne(input.id, session);
     let changes = input;
 
-    // financialReportingType should be subset of its Partner's financialReportingTypes
     const partner = await this.partnerService.readOne(
       object.partner.value!,
       session
     );
-    this.assertFinancialReportingType(
-      input.financialReportingType,
-      partner.financialReportingTypes.value
-    );
-
-    if (
-      !this.validateFinancialReportingType(
+    try {
+      this.verifyFinancialReportingType(
         input.financialReportingType ?? object.financialReportingType.value,
-        input.types ?? object.types.value
-      )
-    ) {
-      if (input.financialReportingType && input.types) {
-        throw new InputException(
-          'Funding type can only be applied to managing partners',
-          'partnership.financialReportingType'
-        );
+        input.types ?? object.types.value,
+        partner
+      );
+    } catch (e) {
+      if (input.types && !input.financialReportingType) {
+        // If input is removing Managing type and FRT is omitted, help caller
+        // out and just remove FRT as well, instead of throwing error.
+        changes = {
+          ...changes,
+          financialReportingType: null,
+        };
       }
-      changes = {
-        ...changes,
-        financialReportingType: null,
-      };
+      throw e;
```

Comment:
Should you throw an error if managing type is removed and FRT omitted? In that case it's not an invalid operation, just incomplete and you're helping it out above.

Reply:
Ah! yes this should be in an else case. Good catch.

Reply:
Yep. :) Everything else looks solid from what I can tell.

Reply:
Sweet! Fixed! 👍